LMSBorrower/src/main/java/com/lms/LMSBorrower/Controller/BorrowerController.java

```diff
@@ -1,29 +1,113 @@
+		
+		if(exists== true) {
+			exists = borrowerService.ifCardExistsBranch(branchId);
+			
+			if(exists ==true) {
+				exists = borrowerService.ifCardExistsBook(branchId, bookId);
+				
+				if (exists ==true) {
+					borrowerService.writeLoans( cardNo, branchId, bookId);
+					return new ResponseEntity<>("Book checked out", HttpStatus.CREATED);
+					}
+				else {
+					return new ResponseEntity<>("Book id not found", HttpStatus.NOT_FOUND);
+					}
+				}
+			
+			else {
+				return new ResponseEntity<>("Branch id not found", HttpStatus.NOT_FOUND);
+				}
+			}
+		
+		else{
+			return new ResponseEntity<>("Invalid card", HttpStatus.NOT_FOUND);
+			}
+		
 	}
 	
 	
+	//delete record in loans table and update copies
+	@RequestMapping(value ="/LMSBorrower/cardNo/{cardNo}/option/2/branch/{branch}/book/{book}", method = {RequestMethod.PUT,  RequestMethod.GET, RequestMethod.DELETE}) 
+	public ResponseEntity<String> writeReturn(@PathVariable (value="cardNo") int cardNo, @PathVariable(value="branch") int branchId, @PathVariable (value="book") int bookId) throws SQLException {
+		
+		boolean exists = borrowerService.ifCardExistsBorrower(cardNo);
+		
+		if(exists== true) {
+			exists = borrowerService.existsBranchReturn(branchId);
+			if(exists ==true) {
+				exists = borrowerService.existsBookReturn(cardNo, branchId, bookId);
+				
+				if (exists ==true) {
+					borrowerService.writeReturn(cardNo, branchId, bookId);
+					return new ResponseEntity<>("Book returned",HttpStatus.ACCEPTED);
+					}
+				else {
+					return new ResponseEntity<>("Book id not found", HttpStatus.NOT_FOUND);
+					}
+				}
+			
+			else {
+					return new ResponseEntity<>("Branch id not found", HttpStatus.NOT_FOUND);
+				}
+			}
+		
+		
+		else{
+			return new ResponseEntity<>("Invalid card", HttpStatus.NOT_FOUND);
+			}
+		}
 	
+	//checkout show branch
+	@RequestMapping("/LMSBorrower/cardNo/{cardNo}/option/1/branch") 
+	public  List<LibraryBranch> checkoutOption(@PathVariable int cardNo) throws SQLException {
+			return 	borrowerService.displayBranchCheckout();
+	}
 	
+	//checkout show book
+		@RequestMapping("/LMSBorrower/cardNo/{cardNo}/option/1/branch/{branch}/book") 
+		public  List<Book> displayCheckoutBook(@PathVariable (value="cardNo") int cardNo, @PathVariable(value="branch") int branchId) throws SQLException {
+				return 	borrowerService.displayBookCheckout(branchId);
+		}
+		//return show branch
+		@RequestMapping("/LMSBorrower/cardNo/{cardNo}/option/2/branch") 
+		public  List<LibraryBranch> ReturnOption(@PathVariable (value="cardNo") int cardNo) throws SQLException {
+				return 	borrowerService.displayBranchCheckout();
+		}
+		
+		//return show book
+				@RequestMapping("/LMSBorrower/cardNo/{cardNo}/option/2/branch/{branch}/book") 
+				public  List<Book> displayReturnBook(@PathVariable(value="branch") int branchId, @PathVariable (value="cardNo") int cardNo) throws SQLException {
+						return 	borrowerService.displayBookReturn(branchId, cardNo);
```

Comment:
These methods are throwing exceptions. Catch them in your DAO so your do not have to throw them on every method.

Reply:
thank you for bringing that up. will work on it